@xen-orchestra/lite/src/components/RemoteConsole.vue

```diff
@@ -3,48 +3,103 @@
 </template>
 
 <script lang="ts" setup>
-import { onBeforeUnmount, ref, watchEffect } from "vue";
+import { fibonacci } from "iterable-backoff";
+import { computed, onBeforeUnmount, ref, watch, watchEffect } from "vue";
 import VncClient from "@novnc/novnc/core/rfb";
 import { useXenApiStore } from "@/stores/xen-api.store";
 
+const N_TOTAL_TRIES = 8;
+const FIBONACCI_MS_ARRAY: number[] = Array.from(
+  fibonacci().toMs().take(N_TOTAL_TRIES)
+);
+
 const props = defineProps<{
   location: string;
   isConsoleAvailable: boolean;
 }>();
 
 const vmConsoleContainer = ref<HTMLDivElement>();
 const xenApiStore = useXenApiStore();
+const url = computed(() => {
+  if (xenApiStore.currentSessionId === null) {
+    return;
+  }
+  const _url = new URL(props.location);
+  _url.protocol = window.location.protocol === "https:" ? "wss:" : "ws:";
+  _url.searchParams.set("session_id", xenApiStore.currentSessionId);
+  return _url;
+});
+
 let vncClient: VncClient | undefined;
+let nConnectionAttempts = 0;
 
-const clearVncClient = () => {
-  if (vncClient !== undefined) {
-    if (vncClient._rfbConnectionState !== "disconnected") {
-      vncClient.disconnect();
+const handleDisconnectionEvent = () => {
+  clearVncClient();
+
+  if (props.isConsoleAvailable) {
+    nConnectionAttempts++;
+
+    if (nConnectionAttempts > N_TOTAL_TRIES) {
+      console.error(
+        "The number of reconnection attempts has been exceeded for:",
+        props.location
+      );
+      return;
     }
-    vncClient = undefined;
+
+    console.error(
+      `Connection lost for the remote console: ${
+        props.location
+      }. New attempt in ${FIBONACCI_MS_ARRAY[nConnectionAttempts - 1]}ms`
+    );
+    createVncConnection();
   }
 };
+const handleConnectionEvent = () => (nConnectionAttempts = 0);
 
-watchEffect(() => {
-  if (
-    !vmConsoleContainer.value ||
-    !xenApiStore.currentSessionId ||
-    !props.isConsoleAvailable
-  ) {
+const clearVncClient = () => {
+  if (vncClient === undefined) {
     return;
   }
 
-  clearVncClient();
+  vncClient.removeEventListener("disconnect", handleDisconnectionEvent);
+  vncClient.removeEventListener("connect", handleConnectionEvent);
 
-  const url = new URL(props.location);
-  url.protocol = window.location.protocol === "https:" ? "wss:" : "ws:";
-  url.searchParams.set("session_id", xenApiStore.currentSessionId);
+  if (vncClient._rfbConnectionState !== "disconnected") {
+    vncClient.disconnect();
+  }
+
+  vncClient = undefined;
+};
+
+const createVncConnection = async () => {
+  if (nConnectionAttempts !== 0) {
+    await new Promise((resolve) =>
+      setTimeout(resolve, FIBONACCI_MS_ARRAY[nConnectionAttempts - 1])
+    );
```

Comment:
No need to change, but for a shorter version you can use `promiseTimeout`:

```suggestion
    await promiseTimeout(FIBONACCI_MS_ARRAY[nConnectionAttempts - 1]);
```